XenAdmin/Alerts/Types/ClientUpdateAlert.cs

```diff
@@ -146,5 +147,50 @@ public static void DownloadAndInstallNewClient(ClientUpdateAlert updateAlert, IW
                     dlg.ShowDialog(parent);
             }
         }
+
+        public static void DownloadSource(IWin32Window parent)
+        {
+            var saveSourceDialog = new SaveFileDialog();
+            saveSourceDialog.FileName = "XenCenter-source";
+            saveSourceDialog.DefaultExt = "zip";
+            saveSourceDialog.Filter = "Zip files (*.zip)|*.zip|All files (*.*)|*.*";
+            saveSourceDialog.InitialDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),"Downloads");

```

Comment:
- I would use the `BrandingManager`'s version of `XenCenter` to name the file.
- Also the filter should be localised since it's visible to customers. Possibly even the `-source` bit in the file name. If you don't want to localise it, we can remove "zip files" but ideally it would. Maybe check what we do in other similar cases.
- For the `InitialDirectory`, using "Downloads" as a name for the folder doesn't necessarily work for all locales. The server status report uses `Win32.GetKnownFolderPath(Win32.KnownFolders.Downloads);` to fetch the path. While I'm not the biggest fan of calling the win32 APIs directly, we already have the methods so might as well use them.
- I'd also use the object initialiser, I think it looks neater. But that's just preference so fell free to ignore:
```csharp
 var saveSourceDialog = new SaveFileDialog
            {
                FileName = "XenCenter-source",
                DefaultExt = "zip",
                Filter = "Zip files (*.zip)|*.zip|All files (*.*)|*.*",
                InitialDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile),"Downloads")
            };
```